src/engine/source/builder/builders/stageBuilderCheck.cpp

```diff
@@ -103,39 +95,134 @@ base::Expression stageBuilderCheckExpression(const std::any& definition,
             field = term.substr(pos1 + 1, pos2 - pos1 - 1);
             value = term.substr(0, pos1) + term.substr(pos2);
```

Comment:
We don't set valueJson when is a helper function, this results in the operation builder generating a function to check if the field is null and not the desired function. Add tests for this use case and fix the implementation.